sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy

```diff
@@ -15,156 +19,100 @@ import life.qbic.business.subscription.fetch.FetchSubscriberOutput
  * @since: 1.0.0
  */
 
-class CreateNotification implements CreateNotificationInput, FetchSubscriberOutput{
+class CreateNotification implements CreateNotificationInput {
+  
     private final CreateNotificationOutput output
-    private final FetchSubscriberInput fetchSubscriberInput
-    private Map<Subscriber, String> notificationPerSubscriber
+    private final FetchUpdatedDataSource dataSource
+    private List<NotificationContent> notifications
 
-    CreateNotification(FetchSubscriberDataSource dataSource, CreateNotificationOutput output){
+    CreateNotification(FetchUpdatedDataSource dataSource, CreateNotificationOutput output) {
         this.output = output
-        this.fetchSubscriberInput = new FetchSubscriber(dataSource, this)
+        this.dataSource = dataSource
     }
 
     @Override
     void createNotifications(String date) {
-        fetchSubscriberInput.fetchSubscriber(date)
-        //assume notificationPerSubscriber will be filled by the output method fetchedSubscribers of the FetchSubscriber use case
-        output.createdNotifications(notificationPerSubscriber)
+      LocalDate localDate = LocalDate.parse(date)
+      
+      try {
+          //1. get todays notifications
+          Map<String, Status> sampleToStatus = dataSource.getUpdatedSamplesForDay(localDate)
+          //2. get projects
+          Map<String,List<String>> projectsWithSamples = getProjectsWithSamples(sampleToStatus.keySet().asList())
+          //3. get project names
+          Map<String,String> projectsWithTitles = dataSource.fetchProjectsWithTitles()
+          
+          for(String projectCode : projectsWithSamples.keySet()) {
+            List<String> samples = projectsWithSamples.get(projectCode)
+            int failedQCCount = filterSamplesByStatus(samples, sampleToStatus, "SAMPLE_QC_FAIL").size()
+            int availableDataCount = filterSamplesByStatus(samples, sampleToStatus, "DATA_AVAILABLE").size()
+            String title = projectsWithTitles.get(projectCode)
+            
+            //4. get subscribers of this projects
+            List<Subscriber> subscribers = getSubscriberForProject(projectCode)
+            
+            for(Subscriber subscriber : subscribers) {
+              notifications.add(new NotificationContent.Builder(subscriber.firstName, subscriber.lastName, subscriber.email, 
+              title, projectCode, failedQCCount, availableDataCount).build())
+            }
+          }
+          output.createdNotifications(notifications)    
```

Comment:
This looks very complex. You could improve code readability if you extract this into another function. 
Effectively you collect the information for each project here, right?
You could introduce a method that generates the notification for a given project, and call this method for each project.

Also the association with the subscriber should be done outside of this method. If you would like to discuss some more detaild suggestions let me know :)